### PR TITLE
Docs: README, Elasticsearch setup, Internal IP setup and Manual steps needed in the Project

### PR DESCRIPTION
**Pull request by dikshamodi**

### CODE REVIEW

**Comment by dikshamodi** [comment]
> Would be great if one of the docs is named `README.md`, so the reader knows which doc to read first

README is a separate doc which I am yet to add. See TODO list `GCP deploy + README` task

**Comment by onish-garg** [comment]
> > Would be great if one of the docs is named `README.md`, so the reader knows which doc to read first
> 
> README is a separate doc which I am yet to add. See TODO list `GCP deploy + README` task

Oh alright - sounds good

File: docs/internalIPsetup.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+CloudSQL and SearchIndex (Elasticsearch run on a compute Engine) can have both internal IP as well as external IP. An external IP is visible outside the cloud environment whereas an internal IP is not accessible outside the cloud environment. 
+
+In an ordinary case, one is able to connect via both Internal IP as well as External IP. 
+
+However, it is not possible to connect to the search Index via external IP. To connect to the search Index via an external IP, we need to ensure that the Search Index server is able to listen to incoming requests on port 9200 and 9300. Doing so would require manually setting up a firewall rule to open these ports to HTTP requests. However, as per [policy](https://g3doc.corp.google.com/company/teams/eip-cloud/gce_enforcer/faq.md?cl=head) the firewall rule would be deleted later. 
+
+To workaround this, we can connect via Internal IP address since connecting via an internal IP does not require a firewall rule to be manually created. A VPC connector has been used for this purpose.
```
Comment (nagasrinivas-google) [inline docs/internalIPsetup.md:11]: How many vpc connectors do we have in total? What exactly each vpc connector is doing?

Comment (dikshamodi) [reply docs/internalIPsetup.md]: We have just one connector. 

The same connector is used for connecting in all cases: 
1. App Engine to CloudSQL
2. App Engine to SearchIndex
3. Cloud Function to SearchIndex

**Pull request merged**